src/components/FilterList/SearchResults/SearchResultsVirtualList.tsx

```diff
@@ -0,0 +1,32 @@
+import { SearchResultsProps } from '@/types/types';
+import { VirtualList } from '../../VirtualList/VirtualList';
+
+import { useSearch } from './hooks';
+
+interface VirtualizedProps<T> extends SearchResultsProps<T> {
+  gap?: number; // Gap in pixels between items
+  overscan?: number; // How many items to preload when scrolling
+}
+
+export function SearchResultsVirtualList<T>({
+  data,
+  renderElement,
+  matcherFunc,
+  gap,
+  overscan
+}: VirtualizedProps<T>) {
+  const filteredElements = useSearch(matcherFunc, data);
+
+  if (filteredElements.length === 0) {
+    return <>No matches found. Try another search</>;
+  }
```

Comment:
Can this check be moved to the `SearchResults` component?  It should be done for the Paginated list as well (and the grid, once that exist)

Reply:
Or should this be an optional component that can be added as a child to SearchResults? :)  Would match the "composite component" design style quite well, maybe?